capstone/src/main/java/com/google/sps/search/Trie.java

```diff
@@ -93,6 +96,140 @@ public void insert(String name, String fullName) {
+      String firstName = input.substring(0, i);
+      String lastName = input.substring(i);
+      String fullName = "";
+      if (reversed) {
+        fullName = lastName + " " + firstName;
+      } else {
+        fullName = firstName + " " + lastName;
+      }
+      if (this.hasFullName(firstName, fullName)) {
+        fullNames.add(fullName);
+      }
+    }
+    return fullNames;
+  }
+
+  /**
+   * Finds names that are within the maximum LED distance of input by calling a recursive helper
+   * method that uses dynamic programming to find the edit distance between words.
+   *
+   * @param input user input
+   * @return Map from possible names to its edit distance from input
+   */
+  public Map<String, Integer> searchLed(String input) {
+    input = input.toUpperCase();
+    Map<String, Integer> suggestionsMap = new HashMap<String, Integer>();
+    int inputLength = input.length();
+    List<Integer> currRow = new ArrayList<Integer>();
+
+    for (int i = 0; i <= inputLength; i++) {
+      currRow.add(i);
+    }
+
+    Map<String, Integer> suggestedNames = new HashMap<String, Integer>();
+    for (Character c : this.children.keySet()) {
+      this.searchLedRecursive(this.children.get(c), c, input, currRow, suggestionsMap);
+      suggestedNames.putAll(suggestionsMap);
+    }
+
+    return suggestedNames;
+  }
+
+  /**
+   * Recursive helper method that populates the new row with values indicating edit distance with
+   * input name. These values are based off of the subproblem (the review row that is passed in).
+   *
+   * @param currNode current trie
+   * @param currChar current character
+   * @param input user input
+   * @param prevRow previous row in matrix
+   * @param suggestions map of suggested names
+   */
+  private void searchLedRecursive(
+      Trie currNode,
+      Character currChar,
+      String input,
+      List<Integer> prevRow,
+      Map<String, Integer> suggestions) {
+    int numColumns = input.length() + 1;
+    List<Integer> currRow = new ArrayList<Integer>();
+    currRow.add(prevRow.get(0) + 1);
+
+    for (int j = 1; j < numColumns; j++) {
+      if (input.charAt(j - 1) == currChar) {
+        currRow.add(prevRow.get(j - 1));
+      } else {
+        currRow.add(1 + Math.min(Math.min(prevRow.get(j), currRow.get(j - 1)), prevRow.get(j - 1)));
+      }
+    }
+
+    /*
+     * If current row indicates end of name and is less than or equal to max LED, add all full names
+     * to map of suggestions and indicate edit distance from input.
+     */
+    int ledDistance = currRow.get(currRow.size() - 1);
+    if (ledDistance <= MAX_LED && currNode.getIsName()) {
+      for (String name : currNode.fullNames) {
+        suggestions.put(name, ledDistance);
+      }
+    }
+
+    /*
+     * If any entry in the current row are less than or equal to the maximum allowed LED
+     * distance, recursively search each branch of the trie from current character.
+     */
+    int minRowVal = Collections.min(currRow);
```

Comment:
`minRowVal` is also not as helpful, `minEditDistance` or `minEditDistanceForRow` or equivalent?